File: apps/actor-ios/ActorClient/CocoaLogProvider.swift
//
//  Copyright (c) 2015 Actor LLC. <https://actor.im>
//

import Foundation

@objc class CocoaLogProvider : NSObject, AMLogProvider {
    
    func warringWithTag(tag: String!, withMessage message: String!) {
        NSLog("⚠️ %@: %@", tag, message);
    }
    
    func verboseWithTag(tag: String!, withMessage message: String!) {
        NSLog("[V] %@: %@", tag, message);
    }
    
    func errorWithTag(tag: String!, withThrowable throwable: JavaLangThrowable!) {
        NSLog("🔴 %@: %@", tag, throwable);
    }
    
    func debugWithTag(tag: String!, withMessage message: String!) {
        NSLog("[D] %@: %@", tag, message);
    }
}

File: actor-sdk/sdk-core/runtime/runtime-cocoa/src/main/java/im/actor/runtime/cocoa/CocoaLogProvider.java
package im.actor.runtime.cocoa;

import im.actor.runtime.LogRuntime;

public class CocoaLogProvider implements LogRuntime {

    @Override
    public native void w(String tag, String message)/*-[
        NSLog(@"WARRING %@: %@", tag, message);
    ]-*/;

    @Override
    public native void e(String tag, Throwable throwable)/*-[
        NSLog(@"ERROR %@:", tag);
        [throwable printStackTrace];
    ]-*/;


    @Override
    public native void d(String tag, String message)/*-[
        NSLog(@"%@: %@", tag, message);
    ]-*/;

    @Override
    public native void v(String tag, String message)/*-[
        NSLog(@"%@: %@", tag, message);
    ]-*/;
}
